Address intermittent test failures

labkey-nicka opened the pull request:
#### Rationale
This PR addresses a couple of issues that are causing intermittent failures in our Jest tests. Namely:

![image](https://user-images.githubusercontent.com/3926239/131715980-e9d37a3f-b3f9-42e3-baae-1d9300e794c4.png)

![image](https://user-images.githubusercontent.com/3926239/131716050-cf1bcd0e-9c24-4a70-9970-bc81c382e314.png)

In both cases I addressed them fixing/hardening the component code. Unfortunately, I was unable to address the intermittent failures we're seeing with the `withQueryModel.spec.tsx` as I'm unable to get them to reproduce.

#### Changes
* Don't presume `reason` exists for errors in `ConceptPathInfo`.
* Prevent state updates in `QuerySelect` after unmount. Improve error handling for failed search requests.

labkey-nicka committed:
Fix flaky tests

labkey-nicka committed:
QuerySelect: Prevent late async setState

cnathe commented on packages/components/src/internal/components/forms/QuerySelect.tsx:
```diff
@@ -183,9 +183,14 @@ export class QuerySelect extends PureComponent<QuerySelectOwnProps, State> {
-            this.setState({ error });
+            if (!this.mounted) return;
+
+            this.setState({ error: error?.message ?? 'Failed to initialize.' });
```
can/should this also use `resolveErrorMessage`?

labkey-nicka replied:
Done.

cnathe reviewed (approved):
Awesome! It will be great to get a couple of these intermittent failures out of the way.
Since this has component changes, don't forget about the release notes update.

labkey-nicka committed:
Use resolveErrorMessage

labkey-nicka committed:
withQueryModels/QuerySelect: override setState

labkey-nicka committed:
Supply id to formsy-react-component
- when labels are present handing in an id avoid JSON serialization of the element (eww)

labkey-nicka committed:
eslint override

labkey-nicka committed:
Fixup some uncaught promises

labkey-nicka committed:
Try out --ci --runInBand

labkey-nicka committed:
Try out --silent

labkey-nicka committed:
Remove setState override

labkey-nicka committed:
Add test-dev target
- will be converted to test once changes are verified

labkey-nicka committed:
ChangePasswordModal: Remove setting "submitting"

labkey-nicka commented:
So I removed the `setState()` override I'd introduced in `withQueryModels` and `QuerySelect` and unfortunately the results remain unstable even with the additional CLI arguments to Jest. Effectively two different error cases pop-up:

1. `TypeError: Cannot read property 'body' of null`. The related React ticket has been closed as "needs more information": https://github.com/facebook/react/issues/15691
![image](https://user-images.githubusercontent.com/3926239/133351946-e56a4251-9331-467a-9530-79252023176e.png)
1. `TypeError: Cannot read property 'createEvent' of null`. A [SO post that talks about](https://stackoverflow.com/questions/60504720/jest-cannot-read-property-createevent-of-null) it or something similar.
![image](https://user-images.githubusercontent.com/3926239/133352133-fdcbe32d-b159-4c60-af08-b58883e2bcab.png)
 
The wild thing about the second one is that it is an error from a component that is not even involved in the test. `QuerySelect` is not invoked anywhere by a `SelectViewInput`, however, `QuerySelect` is somehow being instantiated and then throwing an error. I suspect this is somehow state leaking from another state in the test framework **but** this is even occurring using `--runInBand` which makes me skeptical of even this.

labkey-nicka committed:
Update scripts to "test" and "test-ci"

labkey-nicka committed:
v2.77.1-fb-fix-flaky-jest.0

labkey-nicka committed:
Prepare release notes

labkey-nicka committed:
v2.77.2
